### ISSUE TITLE
Bug: Mullvad suddenly stopped working after a restart, "WARN dns over tls"

### ISSUE THREAD

**Issue by SpellyG** [qdm12/gluetun#899]
### Is this urgent?

No

### Host OS

Synology

### VPN service provider

Mullvad

### What are you using to run the container

Portainer

### What is the version of Gluetun

Running version latest built on 2022-03-15T08:20:32.863Z (commit 984e143)

### What's the problem 🤔

Hello! I've been enjoying gluetun which I run through portainer CE on my Synology NAS for several months.
This week it suddenly stopped working after a restart. now but this week it suddenly stopped working after restart.

I have tried generating new a WireGuard key entirely, using FIREWALL=off, SERVER_HOSTNAME instead of SERVER_CITIES without any luck. I can't get it to run using OpenVPN either.

I seem to be having the same problem as these issues:
#859 
#634 

### Share your logs

Wireguard log:
```log
2022/03/16 21:34:34 INFO routing: default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.2
2022/03/16 21:34:34 INFO routing: local ethernet link found: eth0
2022/03/16 21:34:34 INFO routing: local ipnet found: 172.18.0.0/16
2022/03/16 21:34:34 INFO firewall: enabling...
2022/03/16 21:34:34 INFO firewall: enabled successfully
2022/03/16 21:34:34 INFO storage: merging by most recent 11096 hardcoded servers and 11096 servers read from /gluetun/servers.json
2022/03/16 21:34:34 INFO Alpine version: 3.15.0
2022/03/16 21:34:35 INFO OpenVPN 2.4 version: 2.4.11
2022/03/16 21:34:35 INFO OpenVPN 2.5 version: 2.5.4
2022/03/16 21:34:35 INFO Unbound version: 1.13.2
2022/03/16 21:34:35 INFO IPtables version: v1.8.7
2022/03/16 21:34:35 INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: mullvad
|   |   └── Server selection settings:
|   |       ├── VPN type: wireguard
|   |       ├── Cities: stockholm
|   |       └── Wireguard selection settings:
|   └── Wireguard settings:
|       ├── Private key: *REMOVED*
|       ├── Interface addresses:
|       |   └── *REMOVED*
|       └── Network interface: tun0
├── DNS settings:
|   ├── DNS server address to use: 127.0.0.1
|   ├── Keep existing nameserver(s): no
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       |       ├── 0.0.0.0/0
|       |       └── ::/0
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:7f00:1/104
|               ├── ::ffff:a00:0/104
|               ├── ::ffff:a9fe:0/112
|               ├── ::ffff:ac10:0/108
|               └── ::ffff:c0a8:0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Address to ping: 1.1.1.1
|   └── VPN wait durations:
|       ├── Initial duration: 5s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   ├── Process GID: 1000
|   └── Timezone: Europe/Stockholm
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2022/03/16 21:34:35 INFO routing: default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.2
2022/03/16 21:34:35 INFO routing: adding route for 0.0.0.0/0
2022/03/16 21:34:35 INFO firewall: setting allowed subnets...
2022/03/16 21:34:35 INFO routing: default route found: interface eth0, gateway 172.18.0.1 and assigned IP 172.18.0.2
2022/03/16 21:34:35 INFO pprof http server listening on [::]:6060
2022/03/16 21:34:35 INFO dns over tls: using plaintext DNS at address 1.1.1.1
2022/03/16 21:34:35 INFO healthcheck: listening on 127.0.0.1:9999
2022/03/16 21:34:35 INFO http server: listening on :8000
2022/03/16 21:34:35 INFO firewall: allowing VPN connection...
2022/03/16 21:34:35 INFO wireguard: Using userspace implementation since Kernel support does not exist
2022/03/16 21:34:35 INFO wireguard: Connecting to *REMOVED*:51820
2022/03/16 21:34:35 INFO wireguard: Wireguard is up
2022/03/16 21:34:35 INFO dns over tls: downloading DNS over TLS cryptographic files
2022/03/16 21:34:36 INFO healthcheck: healthy!
2022/03/16 21:34:45 WARN dns over tls: cannot update files: Get "https://www.internic.net/domain/named.root": dial tcp: lookup www.internic.net on 1.1.1.1:53: read udp *REMOVED*:52996->1.1.1.1:53: i/o timeout
2022/03/16 21:34:45 INFO dns over tls: attempting restart in 10s
2022/03/16 21:34:55 INFO dns over tls: downloading DNS over TLS cryptographic files
2022/03/16 21:34:55 ERROR vpn: cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/commits": dial tcp: lookup api.github.com on 1.1.1.1:53: read udp *REMOVED*:46347->1.1.1.1:53: i/o timeout
2022/03/16 21:34:55 ERROR ip getter: Get "https://ifconfig.io/ip": dial tcp: lookup ifconfig.io on 1.1.1.1:53: read udp *REMOVED*:48291->1.1.1.1:53: i/o timeout
2022/03/16 21:34:55 INFO ip getter: retrying in 5s
```

OpenVPN log:
```log
2022/03/16 22:07:56 INFO firewall: allowing VPN connection...
2022/03/16 22:07:56 INFO openvpn: OpenVPN 2.5.4 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov 15 2021
2022/03/16 22:07:56 INFO openvpn: library versions: OpenSSL 1.1.1l  24 Aug 2021, LZO 2.10
2022/03/16 22:07:56 INFO openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]185.65.135.148:1194
2022/03/16 22:07:56 INFO openvpn: UDP link local: (not bound)
2022/03/16 22:07:56 INFO openvpn: UDP link remote: [AF_INET]185.65.135.148:1194
2022/03/16 22:08:01 INFO healthcheck: program has been unhealthy for 5s: restarting VPN
2022/03/16 22:08:01 INFO vpn: stopping
2022/03/16 22:08:01 INFO vpn: starting
```

### Share your configuration

```yml
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    environment:
      - VPN_SERVICE_PROVIDER=mullvad
      - VPN_TYPE=wireguard
      - WIREGUARD_PRIVATE_KEY=*REMOVED*
      - WIREGUARD_ADDRESSES=*REMOVED*/32
      - SERVER_CITIES=Stockholm
      - HEALTH_ADDRESS_TO_PING=1.1.1.1
      - TZ=Europe/Stockholm
    volumes:
      - /volume2/docker/gluetun:/gluetun
    devices:
      - /dev/net/tun:/dev/net/tun
```

**Comment by qdm12** [comment]
For Wireguard, it looks like pinging 1.1.1.1 works but then plaintext DNS is blocked on the vpn server side. Maybe try setting another DNS provider like google?

For Openvpn, there is no reply at all for the VPN server so that's likely an outdated VPN server IP address.

Also the variable is SERVER_HOSTNAMES with an S since v3.28.0 I think.

You can check the wiki page about updating servers information to update mullvad servers, that may help.

Sorry I can't be of any further help, but feel free to share additional discoveries.

**Comment by SpellyG** [comment]
Thank you for the quick response! Seems the issue wasn't with gluetun at all but instead it was the firewall in Synology which must have reset somehow during restart/update.

I allowed communication on all ports for the source IP of the docker container and everything works now, problem solved!

**Comment by SpellyG** [comment]
Not really sure if it truly was healthy, it usually showed up as unhealthy in portainer and then after restarting the container it could show up as healthy.

**Comment by qdm12** [comment]
Yep there was definitely a problem with the healthcheck, it's fixed in v3.27.1, v3.28.1 and in the latest image (by e66ab39f59760258bf351dcd5a1613a654678a7e). It now uses a TCP dial to `github.com:443` instead of pinging `github.com`, since the pinging library that was used was sort of broken I think and would return no error or ping packet back.
